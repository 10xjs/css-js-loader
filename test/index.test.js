const cssJsLoader = require('../');

// Create a sample content passed to cssJsLoader
const style = {
  color: '#eee',
  fontWeight: 'thin',
  fontSize: 24,
  textAlign: 'center',
  padding: 18,
};

// Bind functions to global to replicate `this`
global.cssJsLoader = jest.fn(cssJsLoader);
global.cacheable = jest.fn();
global.exec = jest.fn((x) => x);

// Mock a module
jest.mock('loader-utils', () => {
  return {
    getLoaderConfig: jest.fn(),
  };
});

describe('format works with ', () => {
  it('a javascript style', () => {
    expect(global.cssJsLoader(style)).toMatchSnapshot();
  });

  it('an array of javascript styles', () => {
    expect(global.cssJsLoader([style])).toMatchSnapshot();
  });

  it('es6 objects', () => {
    const sampleObject = {TITLE: style};
    // Make the object pass the es6 check
    Object.defineProperty(sampleObject, '__esModule', {
      value: true,
    });
    expect(global.cssJsLoader(sampleObject)).toMatchSnapshot();
  });

  it('stringified class names', () => {
    const stringedClassName = {
      '.blueText': style,
    };
    expect(global.cssJsLoader(stringedClassName)).toMatchSnapshot();
  });
});
