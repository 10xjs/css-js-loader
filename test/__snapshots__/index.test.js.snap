// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css-js-loader: An array of javascript styles are converted 1`] = `
"color: #eee;
font-weight: thin;
font-size: 24px;
text-align: center;
padding: 18px;
"
`;

exports[`css-js-loader: Basic javascript style is converted 1`] = `
"color: #eee;
font-weight: thin;
font-size: 24px;
text-align: center;
padding: 18px;
"
`;

exports[`css-js-loader: Blank style is not modified 1`] = `""`;

exports[`css-js-loader: Stringified class names are converted 1`] = `
".blueText {
  color: #eee;
  font-weight: thin;
  font-size: 24px;
  text-align: center;
  padding: 18px;
}

"
`;

exports[`css-js-loader: Style url attributes are maintained 1`] = `
"background-image: url('http://example.com/bg.jpg');
color: #eee;
font-weight: thin;
font-size: 24px;
text-align: center;
padding: 18px;
"
`;

exports[`css-js-loader: es6 style objects are converted 1`] = `
".TITLE {
  color: #eee;
  font-weight: thin;
  font-size: 24px;
  text-align: center;
  padding: 18px;
}

"
`;
